clients/gitlabrepo/checkruns_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitlabrepo
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+func Test_CheckRuns(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name         string
+		responsePath string
+		want         []clients.CheckRun
+		wantErr      bool
+	}{
+		{
+			name:         "valid checkruns",
+			responsePath: "./testdata/valid-checkruns",
+			want: []clients.CheckRun{
+				{
+					Status:     "pending",
+					URL:        "https://example.com/foo/bar/pipelines/48",
+					Conclusion: "",
+				},
```

Comment:
If we try this with a completed checkrun, such as the one listed in the example [here](https://docs.gitlab.com/ee/api/pipelines.html#get-a-single-pipeline), I don't think the code, as written, would set the `Conclusion` field.

Might be worth filing an issue and expanding the tests. GitLab doesn't have a `conclusion` field, but I think we can parse it to emulate the GitHub equivalent. Might require some discussion in the issue.

Reply:
#3396